src/main/java/com/kaytat/simpleprotocolplayer/WorkerThreadPair.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2011 The Android Open Source Project
+ * Copyright (C) 2014 kaytat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.kaytat.simpleprotocolplayer;
+
+import android.media.AudioFormat;
+import android.media.AudioManager;
+import android.media.AudioTrack;
+import android.os.Handler;
+import android.util.Log;
+import android.widget.Toast;
+
+/**
+ * 
+ * group everything belongs a stream together,makes multi stream easier
+ * including NetworkReadThread BufferToAudioTrackThread and AudioTrack
+ * 
+ */
+public class WorkerThreadPair {
+
+    static class ThreadStoppable extends Thread {
+        boolean running = true;
+
+        public void customStop() {
+            running = false;
+        }
+    }
+
+    private static final String TAG = WorkerThreadPair.class.getSimpleName();
+
+    private BufferToAudioTrackThread audioThread;
+    private NetworkReadThread networkThread;
+    AudioTrack mTrack;
+
+    public WorkerThreadPair(MusicService musicService, String serverAddr,
+            int serverPort, int sample_rate, boolean stereo, int buffer_ms) {
```

Comment:
I missed the assignment. It's used in brokenShutdown. maybe it's not good .I'm trying to move code out of MusicService without too much structure change, so I leave this as it is